Non-interactive mode

TikiTDO opened the pull request:
This PR introduces a batch mode to the generator, which can be used to programmatically create a bunch of files in one go. This can be extremely useful when bootstrapping a system since such an operation often requires creating a lot of files from a list. This approach is also much easier to integrate with an IDE, opening up the potential for an extension that hooks into the generator.

Closes #47

TikiTDO committed:
Batch generate impl

codeBelt commented on src/utilities/GenerateTemplateFiles.batch.test.ts:
```diff
@@ -0,0 +1,55 @@
+import { IConfigItem } from '../index';
+import CaseConverterEnum from '../constants/CaseConverterEnum';
+
+describe.skip('GenerateTemplateFiles - Batch', () => {
```
Should `.skip` be removed?

```suggestion
describe('GenerateTemplateFiles - Batch', () => {
```

TikiTDO replied:
Ah whoops, copypasta derm. Should I unlock the commandLine tests as well?

codeBelt commented on README.md:
````diff
@@ -234,6 +234,52 @@ One Rule: no spaces between the [Replacer Slots](#replacer-slots-or-ireplacerslo
+
+The following example will generate the component, unit tests, and the SCSS module in one do.
+
+```ts
+// generateTemplateFile.ts
+import { generateTemplateFilesCommandLine, CaseConverterEnum } from 'generate-template-files');
````
Let 's make the example JavaScript to be consistent with the others.

Also change `generateTemplateFilesCommandLine` to `generateTemplateFilesBatch`.

```suggestion
```js
// generateTemplateFile.js
import { generateTemplateFilesBatch } from 'generate-template-files');
```

TikiTDO replied:
I switched the example from TS to JS and fixed the import. I also added a `catch` block to make it clear that it returns a promise.

codeBelt commented on README.md:
```diff
@@ -234,6 +234,52 @@ One Rule: no spaces between the [Replacer Slots](#replacer-slots-or-ireplacerslo
+// generateTemplateFile.ts
+import { generateTemplateFilesCommandLine, CaseConverterEnum } from 'generate-template-files');
+
+export const componentWithInterface = async (componentName: string, componentScope: string = "common"): Promise<void> => {
```
```suggestion
export const componentWithInterface = async (componentName, componentScope = "common") => {
```

codeBelt commented on README.md:
```diff
@@ -234,6 +234,52 @@ One Rule: no spaces between the [Replacer Slots](#replacer-slots-or-ireplacerslo
+  await generateTemplateFilesBatch([
+    {
+      option: 'Component',
+      defaultCase: CaseConverterEnum.PascalCase,
```
```suggestion
      defaultCase: '(pascalCase)',
```

codeBelt commented on README.md:
```diff
@@ -234,6 +234,52 @@ One Rule: no spaces between the [Replacer Slots](#replacer-slots-or-ireplacerslo
+      ],
+      output: {
+        path: `./src/component/__scope__(camelCase)`,
+        pathAndFileNameDefaultCase: CaseConverterEnum.PascalCase,
```
```suggestion
        pathAndFileNameDefaultCase: '(pascalCase)',
```

codeBelt commented on README.md:
```diff
@@ -234,6 +234,52 @@ One Rule: no spaces between the [Replacer Slots](#replacer-slots-or-ireplacerslo
+    },
+    {
+      option: 'Component Interface',
+      defaultCase: CaseConverterEnum.PascalCase,
```
```suggestion
      defaultCase: '(pascalCase)',
```

codeBelt commented on README.md:
```diff
@@ -234,6 +234,52 @@ One Rule: no spaces between the [Replacer Slots](#replacer-slots-or-ireplacerslo
+      ],
+      output: {
+        path: `./src/component/__scope__(camelCase)/I__interface__.ts`,
+        pathAndFileNameDefaultCase: CaseConverterEnum.PascalCase,
```
```suggestion
        pathAndFileNameDefaultCase: '(pascalCase)',
```

codeBelt commented on examples/tools/batchGenerate.js:
```diff
@@ -0,0 +1,41 @@
+// generateTemplateFile.js
+const {generateTemplateFilesBatch, CaseConverterEnum} = require('../../dist/generate-template-files.cjs');
+// Note: In your file it will be like this:
+// const {generateTemplateFilesBatch} = require('generate-template-files');
```
```suggestion
// const {generateTemplateFilesBatch, CaseConverterEnum} = require('generate-template-files');
```

codeBelt reviewed (changes requested):
Nice! Thanks for adding this feature.

TikiTDO committed:
PR Feedback

TikiTDO commented:
Right now it's set to run prettier on `.md` files, which breaks the aligning you had for the `Case Converters` section. I've changed it up a bit so the results are still aligned, but the comment starts immediately after the line, since that's where prettier wants it.

TikiTDO commented:
I fixed up both the tests. The `commandLine` test was expecting an thrown error, but it's currently set up to log the error. I update the test to validate the log (and to avoid logging anything to the screen in the process). I'm not sure which of the approaches you prefer so you may want to take a glance.

I also changed the `.editorconfig` file to set TS indentation rule to 2 space in order to match the prettier config.

TikiTDO commented on .editorconfig:
```diff
@@ -4,7 +4,7 @@ root = true
 charset = utf-8
 indent_style = space
-indent_size = 4
+indent_size = 2
```
This was done to match the `.prettierrc` file

codeBelt commented:
@TikiTDO I will review later and try to get this out today.

TikiTDO commented:
Many thanks
